satpy/tests/reader_tests/test_viirs_l2.py

```diff
@@ -0,0 +1,152 @@
+        if file_type.startswith("CLD"):
+            file_content["geolocation_data/latitude"] = DEFAULT_LAT_DATA
+            file_content["geolocation_data/longitude"] = DEFAULT_LON_DATA
+            if file_type == "CLDPRO":
+                file_content["geophysical_data/Cloud_Top_Height"] = DEFAULT_FILE_DATA
+            elif file_type == "CLDMSK":
+                file_content[
+                    "geophysical_data/Clear_Sky_Confidence"
+                ] = DEFAULT_FILE_DATA
+        elif file_type == "AERDB_":
+            file_content["Latitude"] = DEFAULT_LAT_DATA
+            file_content["Longitude"] = DEFAULT_LON_DATA
+            file_content["Angstrom_Exponent_Land_Ocean_Best_Estimate"] = DEFAULT_FILE_DATA
+            file_content["Aerosol_Optical_Thickness_550_Land_Ocean_Best_Estimate"] = DEFAULT_FILE_DATA
+
+
+class TestVIIRSL2FileHandler:
+    """Test VIIRS_L2 Reader."""
+    yaml_file = "viirs_l2.yaml"
+
+    def setup_method(self):
+        """Wrap NetCDF4 file handler with our own fake handler."""
+        from satpy._config import config_search_paths
+        from satpy.readers.viirs_l2 import VIIRSL2FileHandler
+
+        self.reader_configs = config_search_paths(
+            os.path.join("readers", self.yaml_file)
+        )
+        self.p = mock.patch.object(
+            VIIRSL2FileHandler, "__bases__", (FakeNetCDF4FileHandlerVIIRSL2,)
+        )
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def teardown_method(self):
+        """Stop wrapping the NetCDF4 file handler."""
+        self.p.stop()
+
+    @pytest.mark.parametrize(
+        "filename",
+        [
+            ("CLDPROP_L2_VIIRS_SNPP.A2023364.2230.011.2023365115856.nc"),
+            ("CLDMSK_L2_VIIRS_SNPP.A2023364.2230.001.2023365105952.nc"),
+            ("AERDB_L2_VIIRS_SNPP.A2023364.2230.011.2023365113427.nc"),
+        ],
+    )
+    def test_init(self, filename):
+        """Test basic init with no extra parameters."""
+        from satpy.readers import load_reader
+
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([filename])
+        assert len(loadables) == 1
+        r.create_filehandlers(loadables)
+        # make sure we have some files
+        assert r.file_handlers
+
+    def test_load_aerdb(self):
+        """Test Aerdb File Loading."""
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames(
+            ["AERDB_L2_VIIRS_SNPP.A2023364.2230.011.2023365113427.nc"]
+        )
+        r.create_filehandlers(loadables)
+        datasets = r.load(
+            ["Aerosol_Optical_Thickness_550_Land_Ocean_Best_Estimate", "Angstrom_Exponent_Land_Ocean_Best_Estimate"]
+        )
+        assert len(datasets) == 2
+        for d in datasets.values():
+            assert d.shape == DEFAULT_FILE_SHAPE
+            assert d.dims == ("y", "x")
+            assert d.attrs["sensor"] == "viirs"
```

Comment:
The last 3 tests all have the same body. Could you parametrize them? I think it would be `("filename", "variables")`. You could also add checks for dtype which is a relatively new thing we've been striving for. Something like:

```
for d in datasets.values():
    d_np = d.compute()
    assert d.dtype == d_np.dtype
    assert d.dtype == np.float32
```

There are cases where some operations (scaling the data, masking invalid values, etc) where xarray or numpy will cast data to 64-bit floats. For most satellite data 32-bit floats is good enough. Or in other cases like category products (ex. cloud type) we want things to remain integer types. The dtype could be another parameter in the parametrize if needed.

Reply:
Everything is parameterized with dtype check now, all the datasets are of type 32-bit float so that was a simple addition.